### PR TITLE
Add JWT encoder functionality

### PR DESCRIPTION
**Pull request by wetherc**
Refs #75

This adds a function to encode a JWT for a given user session

### CODE REVIEW

**Comment by wetherc** [comment]
As an example, this will output a token similar to
```
eyJhbGciOiJIUzUxMiIsInR5cCI6IkpXVCJ9.eyJpZCI6ImFzZGZhc2RmIiwiZGF0ZUNyZWF0ZWQiOjE3MDM3MTAwNzE4NTIsInVzZXJJZCI6IjRlMzlkNWU0LTdlMTEtNDIxZi05ZjVmLTk3YjA5Zjg1YTgyMCIsImlhdCI6MTcwMzcxMDA3MTg1MiwiZXhwIjoxNzAzNzEwOTcxODUyfQ.TaX7zZPRid3q7nK4xxOFadre8T15mfpS7BSW34erZL0-TkIfxTjzLnL3BluXAQs2sQEEduBt1XwovPgoXXHHyw
```
which you can decode (https://jwt.io) into:
```json
{
  "id": "asdfasdf",
  "dateCreated": 1703710071852,
  "userId": "4e39d5e4-7e11-421f-9f5f-97b09f85a820",
  "iat": 1703710071852,
  "exp": 1703710971852
}
```

**Review by StephenODea54 — APPROVED** [review]
This is fantastic!

I need to read up on tokens so I can have a better mental model of how we use this on the client. Nonetheless, looks great!

**Comment by wetherc** [comment]
> This is fantastic!
> 
> I need to read up on tokens so I can have a better mental model of how we use this on the client. Nonetheless, looks great!

tl;dr of it all is that on the server we establish "this person is who they claim to be", and send a token back to the client with non-private info (in this case, user ID). This part of the token is public and can be decoded by anyone 

We also cryptographically sign that claim and include the signature in the token. This way when we get it back with a client request, we can verify that the public part of the JWT hasn't been tampered with since it should always hash to the same output as our signature.

As long as the signature is valid for the token, we know the user is legit and don't need to do anything else to authenticate them
